pkg/v1/tkr/pkg/registry/client.go

```diff
@@ -42,24 +42,22 @@ func (r *registry) ListImageTags(imageName string) ([]string, error) {
 
 // GetFile gets the file content bundled in the given image:tag.
 // If filename is empty, it will get the first file.
-func (r *registry) GetFile(image, tag, filename string) ([]byte, error) {
-	ref, err := regname.ParseReference(fmt.Sprintf("%s:%s", image, tag), regname.WeakValidation)
+func (r *registry) GetFile(imageWithTag, filename string) ([]byte, error) {
+	ref, err := regname.ParseReference(imageWithTag, regname.WeakValidation)
```

Comment:
**question**: What's driving this change? (Naively, passing in the image and tag and handling the concatenation within the method _seems_ better.)

Reply:
The main reason behind this change is to support `sha256` based images as well. For example, User can also provide the image as `projects.registry.vmware.com/tanzu/tanzu-plugins/standalone-cliplugins@sha256:e78f0c57a68737561b6217dcbe4976b46393e1ee6615f7c957f162ef428d5ac8`.

For this type of images, providing image as `projects.registry.vmware.com/tanzu/tanzu-plugins/standalone-cliplugins@sha256` and tag as `e78f0c57a68737561b6217dcbe4976b46393e1ee6615f7c957f162ef428d5ac8` seems odd and hence the refactoring.